clover_admin.rb

```diff
@@ -144,9 +144,9 @@ def linkify_ubids(body)
     end
   end
 
-  ObjectAction = Data.define(:label, :flash, :params, :action) do
-    def self.define(label, flash, params = {}, &action)
-      new(label, flash, params.dup.freeze, action)
+  ObjectAction = Data.define(:label, :flash, :redirect, :params, :action) do
+    def self.define(label, flash, redirect = nil, params = {}, &action)
```

Comment:
Let's switch this to keyword arguments. Multiple optional arguments where callers have to use explicit `nil` positional argument is a fairly large smell:

```suggestion
    def self.define(label, flash, redirect: nil, params: {}, &action)
```